packages/@ember/-internals/glimmer/lib/syntax/outlet.ts

```diff
@@ -89,6 +89,9 @@ export const outletHelper = internalHelper(
             return model;
           });
 
+          let componentRef = childRefFromParts(outletRef, ['render', 'component']);
+          named['component'] = componentRef;
```

Comment:
Unlike `model` above, this argument doesn't change for the lifetime of this render state. Its lifetime is the same as `outletRef.render.template`, rather than `outletRef.render.model`.

Reply:
If it doesn't change then something like `createConstRef(state.component, '@component')` should do it. It'll result in less bounds tracking for the DOM etc